runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -5,16 +5,560 @@
+
+        XWalkCoreWrapper.reset(mCoreWrapper, this);
+        XWalkCoreWrapper.check();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        mIsShowing = true;
+
+        XWalkCoreWrapper.reset(mCoreWrapper, this);
+        if (!isXWalkReady() && !isShowingDialog() && !mIsInitializingLibrary) {
+            XWalkCoreWrapper.check();
+        }
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        mIsShowing = false;
+    }
 
     @Override
     public Resources getResources() {
         return getApplicationContext().getResources();
     }
+
+    @Override
+    public final void onObjectInitFailed(Object object) {
+        Log.d("XWalkActivity", "Reserve object: " + object.getClass());
+        mReservedObjects.add(object);
+    }
+
+    @Override
+    public final void onMethodCallMissed(ReflectMethod method) {
+        Log.d("XWalkActivity", "Reserve method: " + method.toString());
+        mReservedMethods.add(method);
+    }
+
+    @Override
+    public final void onXWalkLibraryStartupError(LibraryStatus status, Throwable e) {
+        onXWalkStartupError(status, e);
+    }
+
+    @Override
+    public final void onXWalkLibraryRuntimeError(LibraryStatus status, Throwable e) {
+        onXWalkRuntimeError(status, e);
+    }
+
+    @Override
+    public final void onXWalkLibraryCancelled() {
+        finish();
+    }
+
+    @Override
+    public final void onXWalkLibraryMatched() {
+        Log.d("XWalkActivity", "XWalk library matched");
+        XWalkCoreWrapper.init();
+        mCoreWrapper = XWalkCoreWrapper.getInstance();
+        mLibraryDelegate = XWalkLibraryDelegate.newInstance(this, this);
+
+        mIsInitializingLibrary = true;
+        if (!mLibraryDelegate.initLibrary()) {
+            Assert.fail("XWalk library initialization failed");
+        }
+    }
+
+    @Override
+    public final void onXWalkLibraryPrepared() {
+        Log.d("XWalkActivity", "XWalk library prepared");
+        for (Object object = mReservedObjects.poll(); object != null;
+                object = mReservedObjects.poll()) {
+            Log.d("XWalkActivity", "Init reserved objects: " + object.getClass());
+            try {
+                new ReflectMethod(null, object, "reflectionInit").invoke();
+            } catch (RuntimeException e) {
+                Assert.fail("Reflect object initialization failed");
+            }
+        }
+
+        for (ReflectMethod method = mReservedMethods.poll(); method != null;
+                method = mReservedMethods.poll()) {
+            Log.d("XWalkActivity", "Call reserved methods: " + method);
+            Object[] args = method.getArguments();
+            if (args != null) {
+                for (int i = 0; i < args.length; ++i) {
+                    if (args[i] instanceof ReflectMethod) {
+                        args[i] = ((ReflectMethod) args[i]).invokeWithArguments();
+                    }
+                }
+            }
```

Comment:
We need to call "method.invokeWithArguments()" later even if the args is null. So using continue here is inappropriate.